### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by prithagupta**
#10 Changed the default parameters

- Changed the default parameters for FATERanker: set units=2, joint units=2, joint layers=2, set layers=2
- Changed the default parameters for FETARanker, Ranknet and CmpNet: units = 2 and layers = 2 
- Changed the L2 regularization factor to 1e-4 for all neural networks
- As a default parameter removed the regularization for Expected Rank Regression

### CODE REVIEW

File: csrank/tests/test_ranking.py [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
+FETA_RANKER = 'feta_ranker'
+FATE_RANKER = "fate_ranker"
+
+object_rankers = {FETA_RANKER: FETANetwork, RANKNET: RankNet, CMPNET: CmpNet,
+                  ERR: ExpectedRankRegression, RANKSVM: RankSVM,
+                  FATE_RANKER: FATEObjectRanker}
+object_rankers_params = {
+    FETA_RANKER: {"add_zeroth_order_model": True, "optimizer": SGD(lr=1e-3, momentum=0.9, nesterov=True)},
+    RANKNET: {"optimizer": SGD(lr=1e-3, momentum=0.9, nesterov=True)},
+    CMPNET: {"optimizer": SGD(lr=1e-3, momentum=0.9, nesterov=True)},
```
Comment (kiudee) [inline csrank/tests/test_ranking.py]: This looks like we can factor SGD out and create it in advance.

File: csrank/tests/test_ranking.py [hunk 2]
```diff
@@ -55,24 +76,35 @@ def tunable_parameters(cls):
...
+    os.environ["KERAS_BACKEND"] = "tensorflow"
+    np.random.seed(123)
+    for ranker_name in object_rankers.keys():
+        loss = 0.0
+        rtol = 1e-2
+        atol = 1e-8
+        assert object_ranker_fixed(trivial_ranking_problem, ranker_name=ranker_name, loss=loss, rtol=rtol, atol=atol)
+
+
+def object_ranker_fixed(trivial_ranking_problem, ranker_name=FATE_RANKER, loss=0.0, rtol=1e-2,
```
Comment (kiudee) [inline csrank/tests/test_ranking.py]: Good idea, and I think we can even simplify the code more by using [pytest.mark.parametrize](https://docs.pytest.org/en/latest/parametrize.html#parametrizemark).

File: csrank/tests/test_ranking.py [hunk 3]
```diff
@@ -55,24 +76,35 @@ def tunable_parameters(cls):
...
-    pred = fate.predict(x)
-    assert np.all(pred == y)
+    ranker_params = object_rankers_params[ranker_name]
+    ranker_params['n_object_features'] = ranker_params['n_features'] = 1
+    ranker_params['n_objects'] = 5
+    ranker = object_rankers[ranker_name](**ranker_params)
+    ranker.fit(x, y, epochs=50, validation_split=0, verbose=False)
+    pred_scores = ranker.predict_scores(x)
+    pred_loss = zero_one_rank_loss_for_scores_ties_np(y, pred_scores)
+    print("ranker : {} and 0/1 pred loss: {}".format(ranker_name, pred_loss))
```
Comment (kiudee) [inline csrank/tests/test_ranking.py]: We should not use print statements in tests.

File: csrank/tests/test_ranking.py (hunk 1 above)
Comment (prithagupta) [reply csrank/tests/test_ranking.py]: Done

File: csrank/tests/test_ranking.py (hunk 2 above)
Comment (prithagupta) [reply csrank/tests/test_ranking.py]: Done

File: csrank/tests/test_ranking.py (hunk 3 above)
Comment (prithagupta) [reply csrank/tests/test_ranking.py]: Removed

**Pull request merged**